src/main/java/org/springframework/data/couchbase/repository/query/StringBasedN1qlQueryParser.java

```diff
@@ -196,25 +208,68 @@ private boolean checkNotQuoted(String item, int start, int end, List<int[]> quot
 	private JsonValue getPositionalPlaceholderValues(ParameterAccessor accessor) {
 		JsonArray posValues = JsonArray.create();
 		for (Parameter parameter : this.queryMethod.getParameters().getBindableParameters()) {
-			posValues.add(this.couchbaseConverter.convertForWriteIfNeeded(accessor.getBindableValue(parameter.getIndex())));
+			Object value = couchbaseConverter.convertForWriteIfNeeded(accessor.getBindableValue(parameter.getIndex()));
+				try {
```

Comment:
This bit looks overly complex to debug. Also, it should be properly formatted.

Reply:
I've cleaned this up by pulling the nested code in to separate methods. Also found some duplicate code while doing so.